platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/DynamicVersionResolver.java

```diff
@@ -213,6 +207,15 @@ private RepositoryChainModuleResolution findLatestModule(LinkedList<RepositoryRe
         return best;
     }
 
+    private static void handleFailure(List<RepositoryResolveState> queue, RepositoryResolveState request, RepositoryFailureCollector failures, Exception t) {
+        failures.addFailure(t);
+        if (!request.isContinueOnConnectionFailure() && request.isRepositoryDisabled()) {
```

Comment:
This is a nit, but thinking about these conditions in the opposite order might be better.

e.g., "if (the repository is disabled AND we can't continue to the next one)" vs "if (we can't continue to the next repository AND the repository is disabled)"